tests/test_utils.py

```diff
@@ -82,6 +82,38 @@ def test_geojson_validate():
     )
 
 
+def test_geojson_validate_open_ring():
+    geometry = {"type": "Polygon", "coordinates": [[[1, 2], [3, 4], [5, 6], [7, 8]]]}
+    with pytest.raises(TilesetsError) as excinfo:
+        geojson_validate(2, geometry)
+
+    assert (
+        str(excinfo.value)
+        == "Error in feature number 2: Each linear ring must end where it started"
+    )
+
+
+def test_geojson_validate_closed_ring():
+    geometry = {"type": "Polygon", "coordinates": [[[1, 2], [3, 4], [5, 6], [1, 2]]]}
+    assert geojson_validate(2, geometry) is None
+
+
+def test_geojson_validate_closed_ring_correct_winding_order():
+    geometry = {
+        "type": "Polygon",
+        "coordinates": [[[0, 0], [0, 1], [1, 1], [1, 0], [0, 0]]],
+    }
+    assert geojson_validate(2, geometry) is None
+
+
+def test_geojson_validate_closed_ring_incorrect_winding_order():
+    geometry = {
+        "type": "Polygon",
+        "coordinates": [[[0, 0], [1, 0], [0, 1], [1, 1], [0, 0]]],
```

Comment:
This is an hourglass shape, unsure if that is unintentional. Regardless this would be simpler IMO:

```suggestion
        "coordinates": [[[0, 0], [0, 1], [1, 1], [1, 0], [0, 0]]],
```